### PR TITLE
feat(components): update Badge design

### PR DESCRIPTION
**Pull request by connor-baer**
## Purpose

The badges received a [design refresh on mobile](https://www.figma.com/file/aORNKPp6e3xbpkYrkfoVhW/WIP-Circuit-UI-Mobile?node-id=2168%3A5419) to make them more accessible and refine their purpose. This PR applies the same changes to the web.

## Approach and changes

- Update design of the Badge according to the new [design specs](https://www.figma.com/file/aORNKPp6e3xbpkYrkfoVhW/WIP-Circuit-UI-Mobile?node-id=2168%3A5419):

_Before_

Screen Shot 2020-07-06 at 12 27 53

_After_

Screen Shot 2020-07-04 at 18 39 43

- Use dynamic width for long circular Badge: This prevents the content from overflowing the background:

_Before_

Screen Shot 2020-07-06 at 12 28 43

_After_

Screen Shot 2020-07-04 at 18 50 04

- Deprecate the Badge's `onClick` prop: Badges are not meant to be interactive and should only communicate the status of an element. Use the Tag component instead.
- Deprecate the `primary` Badge color: It conflicts with the color of the primary Button variant. Use the "neutral" variant instead.
- Rename the Badge's `color` prop to `variant`: This makes the prop consistent with the Button and List components.

BREAKING CHANGE:
The **Badge** component's `color` prop has been renamed to `variant` (🤖 _badge-variant-enum_)

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

File: src/cli/migrate/badge-variant-enum.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  components.forEach(component => {
+    renameJSXAttribute(j, root, component, 'color', 'variant');
+  });
+
+  return root.toSource();
+};
+
+export default transform;
```
Comment (hleote) [inline src/cli/migrate/badge-variant-enum.ts:37]: Can you explain why we need to do this? 🙂

File: src/components/Badge/Badge.tsx [hunk 2]
```diff
@@ -139,19 +150,41 @@ const clickableStyles = ({
...
+export const Badge = forwardRef((props: BadgeProps, ref: BadgeProps['ref']) => {
+  if (props.onClick) {
+    deprecate(
+      [
+        'The `onClick` prop of the Badge component has been deprecated.',
+        'Badges are not meant to be interactive and should only',
+        'communicate the status of an element.',
+        'Use the Tag component for interactive elements instead.'
+      ].join(' ')
+    );
```
Comment (hleote) [inline src/components/Badge/Badge.tsx:173]: This is printed out into the console?

File: src/cli/migrate/badge-variant-enum.ts (hunk 1 above)
Comment (connor-baer) [reply src/cli/migrate/badge-variant-enum.ts:37]: The migration guide has more context on codemods: https://github.com/sumup-oss/circuit-ui/blob/beta/MIGRATION.md#-codemods

TL;DR: Codemods enable users to apply breaking changes automatically.

File: src/components/Badge/Badge.tsx (hunk 2 above)
Comment (connor-baer) [reply src/components/Badge/Badge.tsx:173]: Yes, the `deprecate` function logs each deprecation message to the console once.

Comment (hleote) [reply src/components/Badge/Badge.tsx:173]: Nice 👍

File: src/cli/migrate/badge-variant-enum.ts (hunk 1 above)
Comment (hleote) [reply src/cli/migrate/badge-variant-enum.ts:37]: That's amazing... 💯 👍 
I'm impressed 🙂 😉
